### PR TITLE
Added OrderTicket UpdateXXX methods

### PR DESCRIPTION
**Pull request by prasadsom**
Adds variations of  update methods

#### Description
Added 4 variations of update methods

#### Related Issue
#3868 
#### Motivation and Context
As described in linked issue, client call doesn't have to create `UpdateOrderFields`

#### Requires Documentation Change
NA

#### How Has This Been Tested?
I was checking possibility of adding unit tests. But, this class has tight coupling with other member variables, and could not use mocking. Suggestions welcome! And there is no update unit test existing to base new test cases on.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Orders/OrderTicket.cs [hunk 1]
```diff
@@ -273,6 +273,72 @@ public OrderResponse Update(UpdateOrderFields fields)
...
 
+        /// <summary>
+        /// Submits an <see cref="UpdateOrderRequest"/> with the <see cref="SecurityTransactionManager"/> to update
+        /// the ticket with tag specified in <paramref name="tag"/>
+        /// </summary>
+        /// <param name="tag"></param>
+        /// <returns></returns>
+        public OrderResponse UpdateTag(string tag)
+        {
+            UpdateOrderFields fields = new UpdateOrderFields();
```
Comment (jaredbroad) [inline Common/Orders/OrderTicket.cs]: Please use `var` and inline construction here. Please complete the xml params/results documentation.
```
var fields = new UpdateOrderFields() 
{
        Tag = tag
};
```

File: Common/Orders/OrderTicket.cs [hunk 2]
```diff
@@ -273,6 +273,72 @@ public OrderResponse Update(UpdateOrderFields fields)
...
+            {
+                fields.Tag = tag;
+            }
+            return Update(fields);
+        }
+
+        /// <summary>
+        /// Submits an <see cref="UpdateOrderRequest"/> with the <see cref="SecurityTransactionManager"/> to update
+        /// the ticker with stopprice specified in <paramref name="stopPrice"/> and with tag specified in <paramref name="quantity"/>        /// </summary>
+        /// <param name="stopPrice"></param>
```
Comment (jaredbroad) [inline Common/Orders/OrderTicket.cs]: Weird whitespace

Comment (jaredbroad) [reply Common/Orders/OrderTicket.cs]: Great comments overall

File: Common/Orders/OrderTicket.cs (hunk 1 above)
Comment (prasadsom) [reply Common/Orders/OrderTicket.cs]: Changed.

**Pull request merged**